Core/include/Acts/Utilities/Logger.hpp

```diff
@@ -521,6 +521,13 @@ class Logger {
                                         std::placeholders::_1));
   }
 
+  template <typename Callable>
+  void log(const Logging::Level& lvl, Callable&& callable) const {
+    if (doPrint(lvl)) {
+      callable(log(lvl));
+    }
+  }
+
```

Comment:
I added this as a replacement for one use-case of `debugLog`, which is to easily group logging code that only gets executed if the level is met. Do you think this has similar build-perf implications as `debugLog` @HadrienG2? It doesn't use `std::function`. If necessary, I think I can get rid of this as well.

Reply:
It should have much less build-perf implications:

- `doPrint(lvl)` and `log(lvl)` are not templated on `Callable`, so they are only processed once by the compiler. The only thing that the compiler needs to process on every call is the lambda callable itself and a tiny "if this then that" code snippet that uses it.
- This is used much less frequently than `debugLog` was, so the compiler doesn't have that many lambdas to process.